File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/core/note/NoteDao.java
package coolpharaoh.tee.speicher.tea.timer.core.note;


import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;


@Dao
public interface NoteDao {
    @Insert
    void insert(Note items);

    @Update
    void update(Note items);

    @Query("SELECT * FROM note")
    List<Note> getNotes();

    @Query("SELECT * FROM note WHERE tea_id = :teaId")
    Note getNotesByTeaId(long teaId);

    @Query("SELECT * FROM note WHERE tea_id = :teaId AND position = :position")
    Note getNoteByTeaIdAndPosition(long teaId, int position);

    @Query("SELECT * FROM note WHERE tea_id = :teaId AND position >= 0 ORDER BY position")
    List<Note> getNotesByTeaIdAndPositionBiggerZero(long teaId);

    @Query("DELETE FROM note WHERE tea_id = :teaId AND position = :position")
    void deleteNoteByTeaIdAndPosition(long teaId, int position);
}

File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/core/note/NoteDao.kt
package coolpharaoh.tee.speicher.tea.timer.core.note

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface NoteDao {
    @Insert
    fun insert(items: Note)

    @Update
    fun update(items: Note)

    @get:Query("SELECT * FROM note")
    val notes: List<Note>

    @Query("SELECT * FROM note WHERE tea_id = :teaId AND position = :position")
    fun getNoteByTeaIdAndPosition(teaId: Long, position: Int): Note?

    @Query("SELECT * FROM note WHERE tea_id = :teaId AND position >= 0 ORDER BY position")
    fun getNotesByTeaIdAndPositionBiggerZero(teaId: Long): List<Note>

    @Query("DELETE FROM note WHERE tea_id = :teaId AND position = :position")
    fun deleteNoteByTeaIdAndPosition(teaId: Long, position: Int)
}